File: problem_22.rb
#Working out the alphabetical value for each name, multiply this value by its alphabetical position in the list to obtain a name score. What is the total of all the name scores in the file?
class NamesScores

  private

  attr_writer :list_of_names

  def initialize(file)
    self.list_of_names = File.read(file).tr('"', '').split(',').sort
  end

  def alphabetical_value(name)
    name.codepoints.sum { |x| x - 64 }
  end

  public

  attr_reader :list_of_names

  def scores
    list_of_names.sum do |name|
      (list_of_names.index(name) + 1) * alphabetical_value(name)
    end
  end

end

if __FILE__ == $PROGRAM_NAME
  puts NamesScores.new('0022_names.txt').scores
end

File: problem_22.py
def main(file):
    with open(file) as file:
        list_of_names = sorted(file.read().translate(str.maketrans('', '', '"')).split(','))
    total = 0
    for name in list_of_names:
        total += (list_of_names.index(name) + 1) * alphabetical_value(name)
    print(total)

def alphabetical_value(name):
    value = 0
    for char in name:
        value += ord(char) - 64
    return value

if __name__ == "__main__":
    main('0022_names.txt')
